compiler/zrc_typeck/src/tast/ty.rs

```diff
@@ -69,6 +69,11 @@ pub enum Type<'input> {
     Struct(IndexMap<&'input str, Type<'input>>),
     /// Union type literals. Ordered by declaration order.
     Union(IndexMap<&'input str, Type<'input>>),
+    /// Opaque type placeholder used during type resolution for self-referential
+    /// types. This is a temporary type that should be replaced with the
+    /// actual type after the type definition is fully resolved. Opaque
```

Comment:
correct to mention it is replaced with a void pointer `*struct{}`, not with the actual type